### ISSUE TITLE
How to interpret and choose a suitable learning rate when fine-tuning pre-trained model?

### ISSUE THREAD

**Issue by panovr** [facebook/fb.resnet.torch#127]
I'm fine-tuning pre-trained resnet model with custom data set and different learning rates. 
I'm a bit confused about the result of different learning rates.
* with learning rate 0.1
![rest-net-18-lr-0.01](https://2.bp.blogspot.com/-F0eLNWpNmwM/WB8YwwLrseI/AAAAAAAAA3A/GTB_Tqm9JyQ--hKFhwa5bvBoE4Cp5jhAQCLcB/s400/resnet-18.png)
* with learning rate 0.01
![rest-net-18-lr-0.01](https://1.bp.blogspot.com/-tgQABsu0CkE/WB8ZRP-iInI/AAAAAAAAA3E/5cWV9kCZnLwmahBlZ-82iyuzcK3oawS2QCLcB/s400/resnet-18.png)
* with learning rate 0.001
![rest-net-18-lr-0.01](https://1.bp.blogspot.com/-3mr_pKfbPWI/WB8ZtjEAiKI/AAAAAAAAA3I/3q4H-QABNK4cRAO9M2EIcJBmDM6tNnzywCLcB/s400/resnet-18.png)

My questions:
1. How to interpret these results of different learning rates?
2. Which one is better, the one with best error rate?
3. How to a choose suitable learning rate when fine-tuning pre-trained model?

**Comment by colesbury** [comment]
Are these error rates on the training or validation set?

**Comment by panovr** [comment]
These error rates were computed on the validation set.

**Comment by colesbury** [comment]
I think the common practice is to just choose the learning rate schedule which gives the best (lowest) error rate on the validation set. In your case that looks like 0.001.

It might be worth trying to decrease the learning rate even more after the validation error plateaus (around 30-40 epochs in your charts). For example, I might try dropping the learning rate to 1e-4. If you have the checkpoints from epoch 30 or 40 you can just start from there.

**Comment by panovr** [comment]
@colesbury Thanks, and I will do the experiments again.

**Comment by panovr** [comment]
@colesbury I use this command for training 40 epochs with learning rate 0.001:
` th main.lua -resetClassifier true -nClasses 80 -retrain pretrained/resnet-18.t7 -nEpochs 40 -data ~/datasets/data/ -LR 0.001
`
Now, I want to change the learning rate to 0.0001, how can I resume training from the 40th epoch checkpoints? Do I also need the pretrained model?
Thanks!
